CppSource/Include/xsapi/achievements_c.h

```diff
@@ -0,0 +1,235 @@
+
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_MEDIA_ASSET;
+
+
+typedef struct XSAPI_ACHIEVEMENT_REWARD
+{
+    PCSTR name;
+    PCSTR description;
+    PCSTR value;
+    XSAPI_ACHIEVEMENT_REWARD_TYPE rewardType;
+    PCSTR valueType;
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET* mediaAsset;
+
+    XSAPI_ACHIEVEMENT_REWARD_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT_REWARD;
+
+typedef struct XSAPI_ACHIEVEMENT
+{
+    PCSTR id;
+    PCSTR serviceConfigurationId;
+    PCSTR name;
+    XSAPI_ACHIEVEMENT_TITLE_ASSOCIATION** titleAssociations;
+    uint32_t titleAssociationsCount;
+    XSAPI_ACHIEVEMENT_PROGRESS_STATE progressState;
+    XSAPI_ACHIEVEMENT_PROGRESSION progression;
+    XSAPI_ACHIEVEMENT_MEDIA_ASSET** mediaAssets;
+    uint32_t mediaAssetsCount;
+    PCSTR* platformsAvailableOn;
+    uint32_t platformsAvailableOnCount;
+    bool isSecret;
+    PCSTR unlockedDescription;
+    PCSTR lockedDescription;
+    PCSTR productId;
+    XSAPI_ACHIEVEMENT_TYPE type;
+    XSAPI_ACHIEVEMENT_PARTICIPATION_TYPE participationType;
+    XSAPI_ACHIEVEMENT_TIME_WINDOW* available;
+    XSAPI_ACHIEVEMENT_REWARD** rewards;
+    uint32_t rewardsCount;
+    uint64_t estimatedUnlockTime;
+    PCSTR deepLink;
+    bool isRevoked;
+
+    XSAPI_ACHIEVEMENT_IMPL* pImpl;
+} XSAPI_ACHIEVEMENT;
+
+typedef struct XSAPI_ACHIEVEMENTS_RESULT
+{
+    XSAPI_ACHIEVEMENT** items;
+    uint32_t itemsCount;
+    bool hasNext;
+
+    XSAPI_ACHIEVEMENTS_RESULT_IMPL* pImpl;
+} XSAPI_ACHIEVEMENTS_RESULT;
+
+XSAPI_DLLEXPORT bool XBL_CALLING_CONV
+AchievementsResultHasNext(
+    _In_ XSAPI_ACHIEVEMENTS_RESULT* achievementsResult
+) XSAPI_NOEXCEPT;
+
+typedef void(*XSAPI_GET_NEXT_COMPLETION_ROUTINE)(
+    _In_ XSAPI_RESULT_INFO result,
+    _In_opt_ void* context
+    );
+XSAPI_DLLEXPORT XSAPI_RESULT XBL_CALLING_CONV
+AchievementsResultGetNext(
+    _In_ XSAPI_ACHIEVEMENTS_RESULT* achievementsResult,
+    _In_ uint32_t maxItems,
+    _In_ XSAPI_GET_NEXT_COMPLETION_ROUTINE completionRoutine,
+    _In_opt_ void* completionRoutineContext,
+    _In_ uint64_t taskGroupId
+) XSAPI_NOEXCEPT;
+
+
+typedef void(*XSAPI_UPDATE_ACVHIEVEMENT_COMPLETION_ROUTINE)(
+    _In_ XSAPI_RESULT_INFO result,
+    _In_opt_ void* context
+    );
+XSAPI_DLLEXPORT XSAPI_RESULT XBL_CALLING_CONV
+AchievementServiceUpdateAchievement(
+    _In_ XSAPI_XBOX_LIVE_CONTEXT* pContext,
+    _In_ PCSTR xboxUserId,
+    _In_opt_ uint32_t titleId,
+    _In_opt_ PCSTR serviceConfigurationId,
+    _In_ PCSTR achievementId,
+    _In_ uint32_t percentComplete,
+    _In_ XSAPI_UPDATE_ACVHIEVEMENT_COMPLETION_ROUTINE completionRoutine,
+    _In_opt_ void* completionRoutineContext,
+    _In_ uint64_t taskGroupId
+    ) XSAPI_NOEXCEPT;
+
+typedef void(*XSAPI_GET_ACHIEVEMENTS_FOR_TITLE_ID_COMPLETION_ROUTINE)(
+    _In_ XSAPI_RESULT_INFO result,
+    XSAPI_ACHIEVEMENTS_RESULT* achievementsResult,
+    _In_opt_ void* context
+    );
+XSAPI_DLLEXPORT XSAPI_RESULT XBL_CALLING_CONV
+AchievementServiceGetAchievementsForTitleId(
```

Comment:
AchievementService* is a little inconsistent with the naming I have been using.  For example for Title storage my APIs are just TitleStorage* and not TitleStorageService*.  Its fine for now but when we finalize these API's we should be consistent on the naming conventions.

Reply:
AchievementService is the name of the c++ and winrt class. Those methods are the methods of that class.